content/en/blog/_posts/2023-xx-xx-openapi-v3-field-validation-ga.md

```diff
@@ -0,0 +1,107 @@
+that can represent multiple types is also expressed correctly in
+OpenAPI V3 with the oneOf field. This includes proper representations
+for IntOrString and Quantity.
+
+#### Custom Resource Definitions
+
+In Kubernetes, Custom Resource Definitions use a structural OpenAPI V3
+schema that cannot be represented as OpenAPI V2 without a loss of
+certain fields. Some of these include nullable, default, anyOf, oneOf,
+not, etc. OpenAPI V3 is a completely lossless representation of the
+CustomResourceDefinition structural schema.
+
+### How do I use it?
+
+The OpenAPI V3 root discovery can be found at the `/openapi/v3`
+endpoint of a Kubernetes cluster. OpenAPI V3 documents are grouped by
+group-version to reduce the size of the data transported, the separate
+documents can be accessed at `/openapi/v3/apis/<group>/<version>` and
+`/openapi/v3/api/v1` representing the legacy group version. Please
+refer to the [Kubernetes API
+Documentation](https://kubernetes.io/docs/concepts/overview/kubernetes-api/)
+for more information around this endpoint.
+
+Various consumers of the OpenAPI have already been updated to consume v3, including the entirety of kubectl, and server side apply. An OpenAPI V3 Golang client is available in client-go. Please refer to the [interface](https://github.com/kubernetes/client-go/blob/release-1.27/openapi3/root.go) for details on usage.
+
+## Server Side Field Validation
+
+The query parameter `fieldValidation` may be used to indicate the level of field validation the server should perform. If the parameter is not passed, server side field validation is in Warn mode by default.
+
+- Strict: Strict field validation, errors on validation failure
+- Warn: Field validation is performed, but errors are exposed as warnings rather than failing the request
+- Ignore: No server side field validation is performed
+
+kubectl will skip client side validation and will automatically use server side field validation in strict mode. Controllers by default use server side field validation in warn mode.
+
+With client side validation, we had to be extra lenient because some fields were missing from OpenAPI V2 and we didn’t want to reject possibly valid objects. This is all fixed in server side validation. Additional documentation may be found [here](https://kubernetes.io/docs/reference/using-api/api-concepts/#field-validation)
+
+## What's next?
+
+With Server Side Field Validation and OpenAPI V3 released as GA, we introduce more accurate representations of Kubernetes resources. It is recommended to use server side field validation over client side, but with OpenAPI V3, clients are free to implement their own validation if necessary (to “shift things left”) and we guarantee a full lossless schema published by OpenAPI.
+
+Some existing efforts will further improve the information available through OpenAPI including CEL validation and admission, along with OpenAPI annotations on built-in types.
```

Comment:
In addition to manual wrappings, I'd suggest we add links to CEL validation and admission.
```suggestion
Some existing efforts will further improve the information available through OpenAPI
including CEL validation and admission, along with OpenAPI annotations on built-in types.
```